Package/Default.sublime-keymap

```diff
@@ -84,14 +84,30 @@
             { "key": "setting.packagedev.syntax.input_helpers" },
         ]
     },
-    {   // [optional] auto-insert newline after double-colon (contexts) and 'captures'
+    {   // [optional] auto-insert newline after double-colon of 'captures'
         "keys": [":"], "command": "insert", "args": { "characters": ":\n" }, "context": [
-            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {0,2}[^\\s:]+$|^ {4,}(- *)?(captures)+$", "match_all": true },
+            { "key": "preceding_text", "operator": "regex_match", "operand": "^ {4,}(- *)?(captures)+$", "match_all": true },
             { "key": "selection_empty", "match_all": true },
             { "key": "selector", "operand": "meta.block.contexts.sublime-syntax" },
             { "key": "setting.packagedev.syntax.input_helpers" },
         ]
     },
+    {   // [optional] trim space after colon and insert newline.
```

Comment:
What would this be used for? In what situations? And why would trimming trailing whitespace by ST not be the better solution?

Reply:
A space is added after colon when completing statements like `push:` or `set:`. That's fine if a context name or list is entered directly in the same line. But as soon as one presses enter to open up a new anonymous context or add a list of contexts to push/set, the space after the colon should be removed.

This may also be achieved via `trim_trailing_whitespace_on_save`, but as this setting is `false` by default, I ended up very often in leaving spaces after colons.

```
  - match: foo
    push: |

  - match: bar
    push:
      |
```

Reply:
Thanks for the clarification.